prow/github/client.go

```diff
@@ -1182,6 +1183,8 @@ func (c *client) doRequest(ctx context.Context, method, path, accept, org string
 	if err != nil {
 		return nil, fmt.Errorf("failed creating new request: %w", err)
 	}
+	req.Header.Set("X-GitHub-Api-Version", githubApiVersion)
```

Comment:
Responding to https://github.com/kubernetes/test-infra/pull/28340#issuecomment-1372618144 here.

For whatever reason, GitHub chose to call the header `X-GitHub-Api-Version` and not `X-Github-Api-Version`. The reason that it is getting changed to the latter is because `Set()` internally canonicalizes the header name ([ref](https://pkg.go.dev/net/http#Header.Set)). For rules of canonicalization, see [here](https://pkg.go.dev/net/textproto#CanonicalMIMEHeaderKey).

According to the docs:
> To use non-canonical keys, assign to the map directly.

So, to make this work, you'll have to do something like:
```suggestion
	req.Header["X-GitHub-Api-Version"] = []string{githubApiVersion}
```

Reply:
thanks @MadhavJivrajani for the inputs :)

Reply:
that's a TIL